iceoryx_hoofs/container/include/iox/uninitialized_array.hpp

```diff
@@ -15,29 +15,30 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
-#define IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
+#ifndef IOX_HOOFS_CONTAINER_UNINITIALIZED_ARRAY_HPP
+#define IOX_HOOFS_CONTAINER_UNINITIALIZED_ARRAY_HPP
 
 #include "iceoryx_hoofs/iceoryx_hoofs_types.hpp"
 
 #include <cstdint>
 
 namespace iox
 {
-namespace containers
-{
 /// @brief struct used as policy parameter in UninitializedArray to wrap an array with all elements zeroed
 /// @tparam ElementType is the array type
 /// @tparam Capacity is the array size
 template <typename ElementType, uint64_t Capacity>
 struct ZeroedBuffer
 {
+    struct alignas(ElementType) element_t
+    {
+        // AXIVION Next Construct AutosarC++19_03-A18.1.1 : required by low level UninitializedArray building block and encapsulated in abstraction
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
+        cxx::byte_t data[sizeof(ElementType)];
```

Comment:
Adding any other fields to the struct will break things. Maybe worth a comment.

I am also wondering whether the zero-initialization braces `{ }` are required here (instead of array of `element_t`).